kube/file-vault/file-vault-deployment.yml

```diff
@@ -0,0 +1,207 @@
+        service: file-vault-{{ .DRONE_SOURCE_BRANCH }}
+        {{ else }}
+        name: file-vault
+        service: file-vault
+        {{ end }}
+    spec:
+      containers:
+        - name: file-vault
+          # v1.1.0
+          image: quay.io/ukhomeofficedigital/file-vault:f4cd1ece1caddc3bfbc8f5db761c593fda79ef12
+          imagePullPolicy: Always
+          resources:
+            limits:
+              memory: 1024Mi
+              cpu: 200m
+            requests:
+              memory: 512Mi
+          envFrom:
+            - configMapRef:
+                {{ if eq .KUBE_NAMESPACE .BRANCH_ENV }}
+                name: {{ .APP_NAME }}-configmap-{{ .DRONE_SOURCE_BRANCH }}
+                {{ else }}
+                name: {{ .APP_NAME }}-configmap
+                {{ end }}
+          env:
+            # because the internal virus scanner has a self signed cert!
+            - name: NODE_TLS_REJECT_UNAUTHORIZED
+              value: "0"
+            - name: FILE_VAULT_URL
+            {{ if eq .KUBE_NAMESPACE .UAT_ENV }}
+              value: https://fv-{{ .APP_NAME }}.uat.sas-notprod.homeoffice.gov.uk
+            {{ else if eq .KUBE_NAMESPACE .BRANCH_ENV }}
+              value: https://fv-{{ .DRONE_SOURCE_BRANCH }}.branch.sas-notprod.homeoffice.gov.uk
+            - name: DEBUG
+              value: "*"
+            {{ end }}
+            - name: PORT
+              value: "3000"
+            - name: CLAMAV_REST_URL
+              value: https://clamav.virus-scan.svc.cluster.local/scan
+            - name: AWS_ACCESS_KEY_ID
+              valueFrom:
+                secretKeyRef:
+                {{ if or (eq .KUBE_NAMESPACE .BRANCH_ENV) (eq .KUBE_NAMESPACE .UAT_ENV) }}
+                  name: epp-notprod-s3
+                  {{ end }}
+                  key: access_key_id
+            - name: AWS_SECRET_ACCESS_KEY
+              valueFrom:
+                secretKeyRef:
+                {{ if or (eq .KUBE_NAMESPACE .BRANCH_ENV) (eq .KUBE_NAMESPACE .UAT_ENV) }}
+                  name: epp-notprod-s3
+                  {{ end }}
+                  key: secret_access_key
+            - name: AWS_KMS_KEY_ID
+              valueFrom:
+                secretKeyRef:
+                {{ if or (eq .KUBE_NAMESPACE .BRANCH_ENV) (eq .KUBE_NAMESPACE .UAT_ENV) }}
+                  name: epp-notprod-s3
+                  {{ end }}
+                  key: kms_key_id
+            - name: AWS_BUCKET
+              valueFrom:
+                secretKeyRef:
+                {{ if or (eq .KUBE_NAMESPACE .BRANCH_ENV) (eq .KUBE_NAMESPACE .UAT_ENV) }}
+                  name: epp-notprod-s3
+                  {{ end }}
+                  key: bucket_name
+            - name: AWS_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: file-vault
+                  # encryption_key is randomly generated password used to encrypt the params that are returned by file-vault
+                  key: encryption_key
```

Comment:
I would recommend renaming secret in all environments to something more generic, for instance `s3-bucket`. 
which would reduce the amount of checks.

Reply:
s3-notprod-bucket but I was thinking we need to be able to identify the bucket with its service? epp-s3-bucket?